cpu/esp32/periph/i2c_hw.c

```diff
@@ -703,6 +703,7 @@ static void _i2c_transfer (i2c_t dev)
     #if FIFO_USED
     /* reset RX FIFO queue */
     _i2c_hw[dev].regs->fifo_conf.rx_fifo_rst = 1;
+    /* cppcheck-suppress redundantAssignment */
```

Comment:
Is this obvious enough, or should I add a comment explaining that cppcheck is only mildly more aware of C syntax than a regex, and hence doesn't understand that two assignments to a variable with a `volatile` qualifier are for the side affect this causes, not because the value written is of actual interest? Maybe in a bit less snarky words?

I always have the feeling that the reasoning in cppcheck suppression are more about explaining why cppcheck is a poor tool, rather than explaining what the code does. And that the end result is less readable code and reduced maintainability ...

Reply:
Perhaps a short statement would be good that it is a desired behavior here to trigger a reset pulse.

Reply:
Please squash directly.